dev/tools/localization/gen_l10n.dart

```diff
@@ -158,6 +208,34 @@ List<String> genMethodParameters(Map<String, dynamic> bundle, String key, String
   return <String>[];
 }
 
+String generateDateFormattingLogic(Map<String, dynamic> bundle, String key) {
+  String result = '';
+  final Map<String, dynamic> attributesMap = bundle['@$key'] as Map<String, dynamic>;
+  if (attributesMap != null && attributesMap.containsKey('placeholders')) {
+    final Map<String, dynamic> placeholders = attributesMap['placeholders'] as Map<String, dynamic>;
+    for (String placeholder in placeholders.keys) {
+      final dynamic value = placeholders[placeholder];
+      if (_isDateParameter(value)) {
```

Comment:
It would be nice to catch placeholders whose type was 'DateTime' but didn't contain a format key. Maybe make _isDateParameter just check ` placeholderValue['type'] == 'DateTime'` and then either special case the null check with its own error or leave things the way they are.

Reply:
Good point! Done.